lib/Semantics/tools.cpp

```diff
@@ -537,6 +537,18 @@ const DeclTypeSpec *FindParentTypeSpec(const Symbol &symbol) {
   return nullptr;
 }
 
+bool IsSequenceType(const DeclTypeSpec *type) {
+  if (!type) {
+    return false;
+  }
```

Comment:
It looks like `type` is never null so it could be a reference. And given that, it might make more sense for it to be a member function of `DeclTypeSpec`, along with `IsPolymorphic`, `IsNumeric`, etc.

Reply:
Good idea.  I'll do it.